src/text_buffer.rs

```diff
@@ -55,5 +59,21 @@ impl TextBuffer {
         }
     }
 
+    pub fn connect_insert_text<F: Fn(&TextBuffer, &TextIter, &str) + 'static>(&self, f: F) -> u64 {
+        unsafe {
+            let f: Box_<Box_<Fn(&TextBuffer, &TextIter, &str) + 'static>> = Box_::new(Box_::new(f));
+            connect(self.to_glib_none().0, "insert-text",
+                transmute(insert_text_trampoline as usize), Box_::into_raw(f) as *mut _)
+        }
+    }
+}
 
+unsafe extern "C" fn insert_text_trampoline(this: *mut ffi::GtkTextBuffer,
+                                            location: *mut ffi::GtkTextIter,
+                                            text: *mut c_char,
+                                            _len: c_int,
+                                            f: glib_ffi::gpointer) {
+    callback_guard!();
+    let f: &Box_<Fn(&TextBuffer, &TextIter, &str) + 'static> = transmute(f);
+    f(&from_glib_none(this), &from_glib_none(location), &String::from_glib_none(text))
```

Comment:
> `str::from_utf8_unchecked(slice::from_raw_parts(ptr as *const u8, len as usize))`

I wouldn't go with the unchecked variant but `str::from_utf8(...).unwrap()` should be okay.